app/pages/project/classify.cjsx

```diff
@@ -142,7 +143,8 @@ module.exports = React.createClass
         sort: 'cellect' unless SKIP_CELLECT
 
       fetchSubjects = apiClient.type('subjects').get(subjectQuery).then (subjects) ->
-        upcomingSubjects.forWorkflow[workflow.id].push subjects...
+        nonLoadedSubjects = (newSubject for newSubject in subjects when newSubject isnt subjectToLoad)
```

Comment:
The plumbing in the JSON API lib between the request and the resulting objects ensures that if a resource comes back that's already in the local cache, the cached object is updated and returned directly. Two `get`s for the same resource will resolve to the same object.